### PR TITLE
Added kiserdesigns keyboard folder and Madeline keyboard

### PR DESCRIPTION
**Pull request by NCKiser**
## Description

Added kiserdesigns keyboard folder and Madeline keyboard. More kiserdesigns keyboards to be added in future PRs.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kiserdesigns/madeline/readme.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Madeline
+
+![Madeline](https://i.imgur.com/MBT5kg2.jpg)
```
Comment (drashna) [inline keyboards/kiserdesigns/madeline/readme.md:3]:
```suggestion
![Madeline](https://i.imgur.com/MBT5kg2h.jpg)
```

File: keyboards/kiserdesigns/madeline/keymaps/default/config.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#define TAP_CODE_DELAY 10
```
Comment (drashna) [inline keyboards/kiserdesigns/madeline/keymaps/default/config.h:18]: this could be moved to qmk.tap_keycode_delay in the info.json file

Comment (NCKiser) [reply keyboards/kiserdesigns/madeline/keymaps/default/config.h:18]: Would this be in a keymap-level info.json? The delay preferred is highly subjective, and would change based on the keymap used.

File: keyboards/kiserdesigns/madeline/madeline.c
Comment (zvecr) [inline keyboards/kiserdesigns/madeline/madeline.c:1]: This file should be removed.

File: keyboards/kiserdesigns/madeline/madeline.h
Comment (zvecr) [inline keyboards/kiserdesigns/madeline/madeline.h:1]: This file should be removed.

File: keyboards/kiserdesigns/madeline/rules.mk [hunk 3]
```diff
@@ -0,0 +1,7 @@
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
```
Comment (zvecr) [inline keyboards/kiserdesigns/madeline/rules.mk:7]:
```suggestion
# This file intentionally left blank
```

File: keyboards/kiserdesigns/madeline/info.json [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    "maintainer": "NCKiser",
+    "bootloader": "rp2040",
+    "bootmagic": {
+        "enabled": true
+    },
+    "diode_direction": "COL2ROW",
+    "encoder": {
+        "enabled": true,
+        "rotary": [
+            { "pin_a": "GP13", "pin_b": "GP15", "resolution": 4 }
```
Comment (zvecr) [inline keyboards/kiserdesigns/madeline/info.json:13]:
```suggestion
            { "pin_a": "GP13", "pin_b": "GP15" }
```

File: keyboards/kiserdesigns/madeline/info.json [hunk 5]
```diff
@@ -0,0 +1,75 @@
+{
+    "keyboard_name": "Madeline",
+    "manufacturer": "rubybuilds",
+    "maintainer": "NCKiser",
+    "bootloader": "rp2040",
+    "bootmagic": {
+        "enabled": true
+    },
```
Comment (zvecr) [inline keyboards/kiserdesigns/madeline/info.json:8]:
```suggestion
    "features": {
        "bootmagic": true,
        "command": false,
        "console": false,
        "extrakey": true,
        "mousekey": true,
        "nkro": true
    },
```

File: keyboards/kiserdesigns/madeline/info.json [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            { "pin_a": "GP13", "pin_b": "GP15", "resolution": 4 }
+        ]
+    },
+    "matrix_pins": {
+        "cols": [ "GP25", "GP26", "GP27", "GP28", "GP29", "GP9", "GP0", "GP1", "GP2", "GP3" ],
+        "rows": [ "GP5", "GP4", "GP12", "GP7" ]
+    },
+    "mouse_key": {
+        "enabled": true
+    },
```
Comment (zvecr) [inline keyboards/kiserdesigns/madeline/info.json:22]:
```suggestion
```

File: keyboards/kiserdesigns/madeline/keymaps/default/config.h
Comment (zvecr) [inline keyboards/kiserdesigns/madeline/keymaps/default/config.h:1]: This file should be removed.

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by NCKiser — COMMENTED** [review]
Changes have been implemented in 85939f3b189d20f70274eee54fab2dfb6a87f0fe

**Review by drashna — APPROVED** [review]

File: keyboards/kiserdesigns/madeline/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    [_NUM_SYM] = LAYOUT(
+        KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0,
+        KC_ESC, KC_TAB, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_DEL, KC_BSPC,
+        KC_LCBR, KC_LBRC, KC_LPRN, KC_UNDS, KC_NO, KC_RPRN, KC_RBRC, KC_RCBR, KC_SCLN, KC_QUOTE,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+    const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (fauxpark) [inline keyboards/kiserdesigns/madeline/keymaps/default/keymap.c:49]:
```suggestion
    const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/kiserdesigns/madeline/info.json
Comment (fauxpark) [inline keyboards/kiserdesigns/madeline/info.json:1]: Please run `qmk format-json` on this file.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kiserdesigns/madeline/info.json
Comment (NCKiser) [reply keyboards/kiserdesigns/madeline/info.json:1]: okay, done

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by NCKiser** [comment]
@zvecr Is there anything that you recommend changing before merging?

**Review by drashna — APPROVED** [review]

**Pull request merged**